src/app/modules/zxing-scanner/zxing-scanner.component.ts

```diff
@@ -89,6 +89,14 @@ export class ZXingScannerComponent implements AfterViewInit, OnDestroy, OnChange
     @Input()
     autofocusEnabled = true;
 
+    /**
+     * Allow start scan or not.
+     */
+    @Input()
+    set torch(on: boolean) {
+        this.codeReader.setTorch(on);
```

Comment:
I understad `codeReader` as a input class and turning torch on as a output, that seems strange to me, I'm thinking in put that logic in another class that both component and reader makes use. 🤔

Reply:
I cannot understand what you want. Do you mean a `[(torch)]` (banana in box) binding for the torch value?

Reply:
I'm talking about the `this.codeReader` object.

Reply:
Now I see your point, I will do that in a later PR. (maybe next week)

Reply:
I'm very curious about this `set <property>` approach. Do you think we can refactor things like `setCodeReaderThrottling` to that? 🤔